refactor/modules/event_detection.py

```diff
@@ -17,19 +21,218 @@
 # }>
 AnomalyScores = Any
 
+
 class EventDetection:
-    def __init__(self) -> None:
+    def __init__(
+        self,
+        records,
+        num_buckets,
+        num_rows,
+        factor,
+        abs_min_max,
+        smoothing_factor,
+        min_count,
+    ) -> None:
         self.numeric_hash_tables = []
         self.categorical_hash_tables = []
-        pass
-
+        self.records = records
```

Comment:
Shouldn't `records` be passed to `stream_data` instead of being an instance variable? This makes it more obvious how the Event Detection class can stream multiple sets of records iteratively